### DISCUSSION TITLE
Unable to connect to Ollama

### DISCUSSION THREAD

**Discussion by tifDev** [open-webui/open-webui#3864] [General]
# Bug Report

## Description

Getting below error:
![image](https://github.com/user-attachments/assets/2fd82e37-98d6-4cc1-aab9-5ce47518513b)

Note: Accessing webui through http://localhost:3000/

**Bug Summary:**
[Provide a brief but clear summary of the bug]

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

Clean all containers so below command doesn't list any container:
`sudo docker ps -a`

Launch the container:
`sudo docker run -d -p 3000:8080 -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --restart always ghcr.io/open-webui/open-webui:main`

When going to settings page the error message appears.

**Expected Behavior:**
Connect to ollama

**Actual Behavior:**
Cannot connect to ollama

## Environment

- **Open WebUI Version:** v0.3.8

- **Ollama (if applicable):** last

- **Operating System:** Ubuntu 20.04
- **Browser (if applicable):** Firefox

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
![image](https://github.com/user-attachments/assets/85ad830e-758f-43cb-8e5f-c60027574cc8)

**Docker Container Logs:**
![image](https://github.com/user-attachments/assets/020718c9-099d-4ec2-bfde-7157ba1250c1)

**Screenshots (if applicable):**
![image](https://github.com/user-attachments/assets/7279802f-ae7a-4196-8eec-f55f786ac4e1)

## Installation Method

Via docker

**Comment by tifDev** [comment]
If I use:

`docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --restart always ghcr.io/open-webui/open-webui:main` 
It doesn't bind the port properly so that's why I use `-p 3000:8080`

More troubleshooting:

1. I've removed all containers and images + clean docker with system prune
2. No ports conflict ( `sudo lsof -i :3000` )
3. Connectivity from within the container is good (`docker exec -it 0206273293d6 /bin/sh` and then `curl http://localhost:8080` )

**Reply by Peter-De-Ath** [reply]
When you are referencing `OLLAMA_BASE_URL=http://127.0.0.1:11434` this will not work when open-webui is running as a docker container as 127.0.0.1 is referring to the open-webui container itself not the host machine.

when adding --network=host this is binding to the host interfaces so will conflict with any other services you have running on 8080 (I believe you mentioned this on discord?) if you would like to use --network=host and already have some other service running on 8080 you can add `-e PORT=3000` then you should be able to access open-webui at http://localhost:3000

I believe this command will work for you.
`docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main`
Make sure to check Ollama API in Admin Setttings -> Connections it should be http://host.docker.internal:11434

Checkout the Docs, everything you need should be in there https://docs.openwebui.com/getting-started/ :)

**Comment by tifDev** [comment]
So it ends up that the application was accessible via `http://localhost:8080` and by using `OLLAMA_API_BASE_URL=http://127.0.0.1:11434/` I was able to make the connection successfull.


But if you replace `--network=host` by `-p 3000:8080` f.e., I get the connection to Ollama error. 
Can you guys manage to fix that?

**Reply by justinh-rahb** [reply]
`http://host.docker.internal:11434`
